src/sentry/search/events/fields.py

```diff
@@ -150,76 +153,168 @@ def project_threshold_config_expression(organization_id, project_ids):
         .values("project_id", "threshold", "metric")
     )
 
-    num_configured = threshold_configs.count()
-    sentry_sdk.set_tag("project_threshold.count", num_configured)
+    transaction_threshold_configs = (
+        ProjectTransactionThresholdOverride.objects.filter(
+            organization_id=organization_id,
+            project_id__in=project_ids,
+        )
+        .order_by("project_id")
+        .values("transaction", "project_id", "threshold", "metric")
+    )
+
+    num_project_thresholds = project_threshold_configs.count()
+    sentry_sdk.set_tag("project_threshold.count", num_project_thresholds)
     sentry_sdk.set_tag(
         "project_threshold.count.grouped",
-        format_grouped_length(num_configured, [10, 100, 250, 500]),
+        format_grouped_length(num_project_thresholds, [10, 100, 250, 500]),
     )
 
-    if num_configured == 0:
+    num_transaction_thresholds = transaction_threshold_configs.count()
+    sentry_sdk.set_tag("txn_threshold.count", num_transaction_thresholds)
+    sentry_sdk.set_tag(
+        "txn_threshold.count.grouped",
+        format_grouped_length(num_transaction_thresholds, [10, 100, 250, 500]),
+    )
+
+    if num_project_thresholds + num_transaction_thresholds == 0:
         return ["tuple", [f"'{DEFAULT_PROJECT_THRESHOLD_METRIC}'", DEFAULT_PROJECT_THRESHOLD]]
-    elif num_configured > MAX_QUERYABLE_TRANSACTION_THRESHOLDS:
+    elif num_project_thresholds + num_transaction_thresholds > MAX_QUERYABLE_TRANSACTION_THRESHOLDS:
         raise InvalidSearchQuery(
             f"Exceeded {MAX_QUERYABLE_TRANSACTION_THRESHOLDS} configured transaction thresholds limit, try with fewer Projects."
         )
 
-    return [
-        "if",
+    project_confiq_query = (
```

Comment:
not sure if intentional:
```suggestion
    project_config_query = (
```